noir/noir-repo/tooling/nargo/src/ops/transform.rs

```diff
@@ -2,16 +2,20 @@ use acvm::acir::circuit::ExpressionWidth;
 use iter_extended::vecmap;
 use noirc_driver::{CompiledContract, CompiledProgram};
 
+/// TODO(https://github.com/noir-lang/noir/issues/4428): Need to update how these passes are run to account for
+/// multiple ACIR functions
+
 pub fn transform_program(
-    mut program: CompiledProgram,
+    mut compiled_program: CompiledProgram,
     expression_width: ExpressionWidth,
 ) -> CompiledProgram {
+    // TODO: Work to get rid of these clones by borrowing `Circuit` in the acvm compiler or by accepted a `Program`
```

Comment:
We should be able to use `std::mem::take` to avoid this.

Reply:
good call done